src/libavrdude.h

```diff
@@ -60,6 +60,35 @@ typedef uint32_t pinmask_t;
 #define LIBAVRDUDE_SOFTFAIL (-3) // returned, eg, by avr_signature() if caller
                                  // might proceed with chip erase
 
+/* Message system */
+/* These 2 functions are supposed to be supplied by the application */
+
+int avrdude_message(int msglvl, const char *format, ...);
+int avrdude_message2(FILE *fp, int lno, const char *file, const char *func, int msgmode, int msglvl, const char *format, ...);
+
+enum msglvl {
+  MSG_EXT_ERROR = (-3), // OS-type error, no -v option, can be suppressed with -qqqqq
```

Comment:
Tiny note: the AVRDUDE project has been placing inline comments that are right of the code from Column 32 onwards, ie, best to move these comments to the right by 8 spaces